### DISCUSSION TITLE
(googlechrome) Select chrome version using chrome XML API

### DISCUSSION THREAD

**Discussion by user-name11111** [chocolatey-community/chocolatey-packages#2054] [Ideas]
Hello! 

How about adding functionality to downgrade to specific version of chrome?

``` 
chrome="8A69D345-D564-463C-AFF1-A69D9E530F96"
update_url = 'https://tools.google.com/service/update2' 
request = <?xml version='1.0' encoding='UTF-8'?>
            <request protocol='3.0' version='1.3.23.9' ismachine='0'
                     installsource='ondemandcheckforupdate' dedup='cr'
                     sessionid='{3597644B-2952-4F92-AE55-D315F45F80A5}'
                     requestid='{CD7523AD-A40D-49F4-AEEF-8C114B804658}'>
            <hw sse='1' sse2='1' sse3='1' ssse3='1' sse41='1' sse42='1' avx='1'/>
            <os platform='win' version='6.3' arch='"""+arch+"""'/>
            <app appid='${chrome}' ap='' version='${version}' nextversion='' lang='' brand='GGLS' client='''>
                <updatecheck/>
            </app>
            </request>
```
```
curl "https://tools.google.com/service/update2" --data "$req"
Then parse XML answer
```

**Comment by AdmiringWorm** [comment]
I'm sorry that this discussion has gone unnoticed. Unfortunately, what you want is impossible for this package when hosted on the Chocolatey Community Repository.

We need to use official download locations for the google chrome package. Since the only one available is for the latest version, we can not provide a downgrade ability or even the ability to install a specific version of google chrome.

It can be done for chromium but not for google chrome. Anyone needing a specific version of Google chrome must host their own internal repository and internalize (either automatically or manually) the package when it is first published to the community repository.
